pkg/rancher-desktop/backend/wsl.ts

```diff
@@ -1175,15 +1175,21 @@ export default class WSLBackend extends events.EventEmitter implements VMBackend
                 await this.writeFile(`/etc/conf.d/buildkitd`, SERVICE_BUILDKITD_CONF);
               }),
               this.progressTracker.action('Configuring image proxy', 50, async() => {
-                const allowListConf = BackendHelper.createImageAllowListConf(config.containerEngine.imageAllowList);
+                const imageAllowListConf = '/usr/local/openresty/nginx/conf/image-allow-list.conf';
                 const resolver = `resolver ${ await this.ipAddress } ipv6=off;\n`;
 
                 await this.writeFile(`/usr/local/openresty/nginx/conf/nginx.conf`, NGINX_CONF, 0o644);
                 await this.writeFile(`/usr/local/openresty/nginx/conf/resolver.conf`, resolver, 0o644);
-                await this.writeFile(`/usr/local/openresty/nginx/conf/image-allow-list.conf`, allowListConf, 0o644);
                 await this.writeFile(`/etc/logrotate.d/openresty`, LOGROTATE_OPENRESTY_SCRIPT, 0o644);
 
                 await this.runInstallScript(CONFIGURE_IMAGE_ALLOW_LIST, 'configure-image-allow-list');
+                if (config.containerEngine.imageAllowList.enabled) {
+                  const patterns = BackendHelper.createImageAllowListConf(config.containerEngine.imageAllowList);
+
+                  await this.writeFile(imageAllowListConf, patterns, 0o644);
+                } else {
+                  await this.execCommand({ root: true }, 'rm', '-f', imageAllowListConf);
```

Comment:
Here, we delete the allow list if it's disabled, so the openresty service will never start correctly. But (see other comment) containerd still needs it.

Reply:
I've removed the `need` changes. We must delete the file because it works as a sentinel to tell the container engine init scripts and the `nerdctl` wrapper to configure `HTTPS_PROXY` (or not, if the file doesn't exist).